src/frontend/src/optimizer/rule/apply_join_transpose_rule.rs

```diff
@@ -120,6 +120,11 @@ impl Rule for ApplyJoinTransposeRule {
             return None;
         }
 
+        assert!(
```

Comment:
Include a comment that `ProjectJoinSeparateRule` is always applied before this rule can match.